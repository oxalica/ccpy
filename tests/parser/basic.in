a
1 + 2 + 3
1 + 2 * ~~3
a + 1 % ...
1,
pass
1, 2, 3
