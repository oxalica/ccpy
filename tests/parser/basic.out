StmtExpr(ExprName("a"))
StmtExpr(ExprBinary("add", ExprBinary("add", ExprLiteral(LitInteger(1)), ExprLiteral(LitInteger(2))), ExprLiteral(LitInteger(3))))
StmtExpr(ExprBinary("add", ExprLiteral(LitInteger(1)), ExprBinary("mul", ExprLiteral(LitInteger(2)), ExprUnary("inv", ExprUnary("inv", ExprLiteral(LitInteger(3)))))))
StmtExpr(ExprBinary("add", ExprName("a"), ExprBinary("mod", ExprLiteral(LitInteger(1)), ExprLiteral(LitEllipse))))
StmtExpr(ExprTuple(ExprLiteral(LitInteger(1))))
StmtExpr(ExprTuple(ExprLiteral(LitInteger(2)), ExprLiteral(LitInteger(3))))
StmtPass
StmtExpr(ExprTuple(ExprLiteral(LitInteger(1)), ExprTuple(ExprLiteral(LitInteger(2))), ExprTuple(ExprLiteral(LitInteger(3)), ExprLiteral(LitInteger(4))), ExprLiteral(LitInteger(5))))
StmtExpr(ExprCall(ExprCall(ExprName("f"), [ExprLiteral(LitInteger(1)), ExprLiteral(LitInteger(2)), ExprBinary("add", ExprLiteral(LitInteger(3)), ExprCall(ExprLiteral(LitInteger(4)), [])), ExprTuple()]), []))
StmtExpr(ExprTuple())
StmtExpr(ExprLiteral(LitBool(true)))
StmtExpr(ExprTuple(ExprLiteral(LitBool(true)), ExprBinary("add", ExprLiteral(LitBool(false)), ExprLiteral(LitNone))))
StmtExpr(ExprTuple(ExprLiteral(LitStr("123")), ExprLiteral(LitStr("234"))))
StmtExpr(ExprTuple(ExprLiteral(LitStr("3\n'")), ExprLiteral(LitStr("\\\""))))
StmtGlobal("a")
StmtNonlocal("b", "c", "d")
StmtReturn(Some(ExprIndex(ExprTuple(ExprLiteral(LitInteger(1))), ExprLiteral(LitInteger(0)))))
StmtExpr(ExprIndex(ExprLiteral(LitStr("123")), ExprTuple(ExprLiteral(LitInteger(1)), ExprLiteral(LitInteger(2)))))
StmtRaise(ExprCall(ExprName("Exception"), []))
StmtAssign([PatName("a")], ExprLiteral(LitInteger(1)))
StmtAssign([PatTuple[PatName("a"), PatName("b")]], ExprTuple(ExprName("b"), ExprName("a")))
StmtAssign([PatName("a"), PatAttr(ExprName("b"), "c"), PatTuple[PatName("d"), PatName("e")], PatIndex(ExprCall(ExprName("f"), []), ExprCall(ExprName("g"), []))], ExprTuple(ExprLiteral(LitInteger(1)), ExprName("a")))
StmtDel(PatName("a"))
StmtDel(PatTuple[PatName("b"), PatName("c")])
StmtDel(PatIndex(ExprName("d"), ExprLiteral(LitInteger(0))))
StmtDef("f", [], None, [StmtPass])
StmtDef("g", [("x", None)], None, [StmtReturn(Some(ExprName("x")))])
StmtDef("h", [], Some("xs"), [StmtReturn(Some(ExprName("xs")))])
StmtDef("i", [("a", None), ("b", Some(ExprBinary("add", ExprLiteral(LitInteger(1)), ExprLiteral(LitInteger(2)))))], Some("xs"), [StmtReturn(None)])
StmtDef("j", [], None, [StmtPass])
StmtDef("k", [], None, [StmtExpr(ExprLiteral(LitInteger(1))), StmtDef("l", [], None, [StmtReturn(None)]), StmtReturn(Some(ExprName("l")))])
End
